### ISSUE TITLE
Test react app feedback.

### ISSUE THREAD

**Issue by keystroke3** [iradualbert/photodino-frontend#1]
Hello, Albert
I am a member of the FotoDino backend team and I am responsible for evaluating your app. 
### The good
Of all candidates so far, your implementation comes closest to what we had in mind when we gave out the task. The app looks relatively better than some of our other responses, and it actually fetches data from the given api. The fuzzy search in the `Locations` is something none of the other candidates implemented.
### The bad
All praise comes with some criticism. Here are some of the things we found poorly implemented or lacking in your app:
#### Creating new data
There is no way to create new cities and locations. We therefor can't be able to evaluate your usage of the `POST` method that the api provides. If there is, then it is very hard to find.
#### Updating existing data
As far as we can tell, there is no way of updating existing data for cities and locations. You can only see the data, but not update any of the fields. We can't evaluate your usage of `PUT/PATCH` methods in the api.
#### Deleting a location/city
For all cities and locations, there is a delete button, that when clicked, removes the entry from the view, but doesn't actually delete the entity. Upon reload, the entities that were supposed to be deleted are not being deleted. The application doesn't seem to be making any `DELETE` calls to the api from the traffic log.
#### No detailed view of a location.
There is a detailed view of a city, but no detailed view of a location. When I click on a location, I would like to be able to see all the location's details, including the id, time created. 
#### No way of going home
If I am viewing the a location and want to go back and view a city, there is no way for me to do that, other than using the browser's history.

### The ugly
You application seems to be facing a problem with an infinite loop or unending recursion when viewing the city details and the locations in the city. This leads to a high cpu usage.
![image](https://user-images.githubusercontent.com/37796862/133982529-a1945f1e-798b-4481-9900-5850184fc3af.png)

You can see that while that tab was open, for just a few seconds, the last error occurred over 100 times. This does not look good for your application. 

### Closing thoughts
We were quite impressed with your implementation, and we are happy to give you another chance at it. If you wish to continue, you can improve the app by fixing the issues mentioned above. You should also switch from list view to grid view due to the large amount of information being shown. A paginated view will also earn you more points and improve your chances. If you wish to give it another try, we will give you 48 hours from the time of writing to make the changes. Once you are done, reply to this issue and tell us the changes you made so we can look at it again. If you don't wish to continue, you can reply to this issue telling us why, so we can remove your name from the shortlist. 
Thank you, 
Ted

**Comment by iradualbert** [comment]
Hello,
Thank you for your feedback.

So here is what I did,

### The infinite Loop

The way I was updating the state in the `useEffect` hook was leading to infinite re-rendering of the component. Now, the page is working fine without the infinite loop.


### Creating, Updating and Deleting Data

Now you can add a new city, update and delete existing cities. Since Creating and Updating locations would follow the same logic, I did not implement those features.  I think they are not necessary for the demonstration purpose.

### Going Home

I have added a navigation bar. You can now easily navigate through 
the pages using the navbar.


### Additional notes

I can not implement pagination since the API does not support it.
In other words, pagination is not necessary because the API returns all data.

**Comment by keystroke3** [comment]
Greetings again,
Thank you for updating your app. I am happy to see you fixed most of the issues I had brought up. I however, don't see an option to update a location. We are currently finalizing the applications, as all the ones who sent their demos and were corrected have made changes as well. We will pick the ones we want to interview and schedule meetings with them. You are not obliged to, but you can add the update option for the locations in the mean time.  
We will get in touch with you when we have finalized. We will notify you if you have been selected or not.
Thank You,
Ted
